packages/menu-bar/src/vaadin-menu-bar-buttons-mixin.js

```diff
@@ -40,6 +50,18 @@ export const ButtonsMixin = (superClass) =>
       super.connectedCallback();
 
       this._initButtonAttrs(this._overflow);
+
+      // Observe parent node resize to detect overflow
+      this.__parent = this.parentNode instanceof ShadowRoot ? this.parentNode.host : this.parentNode;
+      this.__parentResizeObserver.observe(this.__parent);
```

Comment:
Should this be moved to `connectedCallback` instead? If the element is removed and reattached, then the parent will not be observed.

Reply:
This is already in the `connectedCallback`, I can add a test for re-attaching the parent.

Reply:
Sorry, my bad. I was checking only from the GH UI and got fooled by the collapsed lines (I should've notice the `super.connectedCallback()` call) 🤦🏻‍♂️

<img width="441" alt="image" src="https://user-images.githubusercontent.com/262432/167802347-98210380-edeb-4b5c-9960-217e353a9aca.png">